docs/rules/max-classes-per-file.md

```diff
@@ -0,0 +1,39 @@
+# enforce a maximum number of classes per file (max-classes-per-file)
+
+Files containing multiple classes can often result in a less navigable
+and poorly structured codebase. Best practice is to keep each file
+limited to a single responsibility.
+
+## Rule Details
+
+This rule enforces that each file may contain only a particular number
+of classes and no more.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint max-classes-per-file: "error"*/
+
+class Foo {}
+class Bar {}
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint max-classes-per-file: "error"*/
+
+class Foo {}
+```
+
+## Options
+
+This rule has a numeric option (defaulted to 1) to specify the
+maximum number of classes.
```

Comment:
Nitpick: It would be nice to include an example here of what the configuration looks like (e.g. if a user isn't familiar with how rules are configured in general), but it's not a blocker for this PR.